packages/flutter_tools/lib/src/device.dart

```diff
@@ -189,7 +193,21 @@ abstract class Device {
 abstract class DeviceLogReader {
   String get name;
 
-  Future<int> logs({ bool clear: false, bool showPrefix: false });
+  /// A broadcast stream where each element in the string is a line of log
+  /// output.
+  Stream<String> get lines;
```

Comment:
We use onFoo for things that are callbacks. This is a Stream, so "lines" seems fine.